### DISCUSSION TITLE
withRouter and ReactDOM.render

### DISCUSSION THREAD

**Discussion by dvioleau1** [vercel/next.js#13034] [Help] [answered]
Hello.
I use withRouter from next/router but I need to use it with react-dom.
My index.js is :

ReactDOM.render(
<App />,
document.getElementById('root')
);
and at the end of App.js i use :
export default withRouter(Home);
It does not accept withRouter. I really need to use document because of an other component.
Is there a way to use withRouter from next and reactDom.render ?
Thank you in advance

**Comment by jamesmosier** [comment, marked as answer]
`withRouter` uses the React Context API to provide the router state to your component, therefore if the component you are rendering is not within Next's component tree (i.e. a child of the Context provider) then you cannot access that Context's state via `withRouter`.

**Reply by timneutkens** [reply]
To add to that, in general you never need to run `ReactDOM.render` yourself in a Next.js application.
